### ISSUE TITLE
hs.GetFirstLocationList and hs.GetSecondLocationList - tested???

### ISSUE THREAD

**Issue by alexbk66** [HomeSeer/Plugin-SDK#170]
```
Unable to cast object of type 'System.Collections.SortedList' to type 'System.Collections.Generic.IEnumerable 1[System.String]'.
Service Version: NO_VERSION
```
Seriously? I'm wasting my weekend testing your code?  Do you want to hire me as paid QA?

**Comment by alexbk66** [comment]
An should `List<HsDevice> GetAllDevices(bool withFeatures)` return `List<AbstractHsDevice`, not `List<HsDevice>`?

**Comment by spudwebb** [comment]
> An should `List<HsDevice> GetAllDevices(bool withFeatures)` return `List<AbstractHsDevice`, not `List<HsDevice>`?

No, HsDevice has a Features property, so if you pass true to GetAllDevices() you will get the features associated with your HsDevice in this property.

Problem with GetFirstLocationList  and GetSecondLocationList  tracked as HS-1130

**Comment by spudwebb** [comment]
This is fixed in HS 4.1.14.0, GetFirstLocationList  and  GetSecondLocationList  now work as expected
